### PR TITLE
handle NaN reference value

### PR DESCRIPTION
**Pull request by nw13slx**
## Description
Allow NaN as a reference value. These NaN entries will be ignored during mean and sum.

This feature is disabled unless "has_nan" is explicitly defined in the loss function parameters.

## How Has This Been Tested?

Test with two new unit tests in trainer/test_loss.py

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] My code follows the code style of this project and has been formatted using `black`.
- [ ] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] `example.yaml` (and other relevant `configs/`) have been updated with new or changed options.
- [ ] I have updated `CHANGELOG.md`.

### CODE REVIEW

File: nequip/train/_loss.py [hunk 1]
```diff
@@ -73,38 +73,59 @@ def __call__(
...
-            atomic_weight_on = False
+            per_atom_loss = self.func(pred[key], ref[key])
+
+        reduce_dims = tuple(i + 1 for i in range(len(per_atom_loss.shape) - 1))
+
+        if has_nan:
+            if len(reduce_dims) > 0:
+                per_atom_loss = per_atom_loss.sum(dim=reduce_dims)
+
+            spe_idx = pred[AtomicDataDict.SPECIES_INDEX_KEY]
```
Comment (Linux-cpp-lisp) [inline nequip/train/_loss.py:97]: isn't this broken? since SPECIES_INDEX_KEY no longer exiss
